functorch/csrc/init.cpp

```diff
@@ -99,6 +146,14 @@ Tensor _remove_batch_dim(const Tensor& self, int64_t level, int64_t batch_size,
   return result;
 }
 
+Tensor _unwrap_functional_tensor(const Tensor& self) {
+  auto* functional = dynamic_cast<FunctionalTensorWrapper*>(self.unsafeGetTensorImpl());
+  // We only ever call that after popping out of a functionalize() call, in which case the current tensors
+  // should always be wrapped in a FunctionalTensorWrapper.
+  TORCH_INTERNAL_ASSERT(functional != nullptr);
+  return functional->value();
```

Comment:
Do we need to _sync here? Or is it already guaranteed that all the queued updates have been applied?

Reply:
we do, and this actually fixes a bug. Thanks!